### PR TITLE
Implements GET submissions for user profile

### PR DESCRIPTION
**Pull request by perryraskin**
This displays contests under the `Entered Contests` tab on the user profile.

I also fixed a minor thing in `SubmitSubmission.js` - the `fetch` call needed to be a `POST` instead of a `PUT` and the api endpoint was missing the `/submission` at the end. @iglesias294 let me know if all good there.

### CODE REVIEW

**Comment by perryraskin** [comment]
Also, so far this only works by logging in via the `Signup` page, as it passes a `user` object with the user's info and `token` for use in api calls. Once you sign in, it redirects you to the `Profile` page and works as expected. Happy to make changes if there's a better way at this point, or we can wait a bit and refactor when we have more finished?

**Review by iglesias294 — APPROVED** [review]
Looks good to me.

File: server/routes/api/users.js [hunk 1]
```diff
@@ -87,4 +88,37 @@ router.get("/contests", (req, res) => {
...
+router.get("/submissions", auth, (req, res) => {
+  const { user_id } = req.query;
+  let result = {};
+
+  Submission.find({ user_id })
+    .then((submissions) => {
+      let contestIds = [];
+      submissions.forEach(submission => contestIds.push(submission.contest_id));
+      result.submissions = submissions;
+      Contest.find({
```
Comment (bonnieli) [inline server/routes/api/users.js:103]: i wonder if you can use an aggregator to simplify this query?

https://docs.mongodb.com/manual/reference/operator/aggregation/lookup/

**Review by bonnieli — COMMENTED** [review]
just a comment on aggregation , but FE looks good

**Pull request merged**